backend/pvm.pm

```diff
@@ -0,0 +1,151 @@
+package backend::pvm;
+use strict;
+use base ('backend::baseclass');
+use bmwqemu qw(fileContent diag save_vars);
+require IPC::System::Simple;
+use autodie qw(:all);
+use File::Basename;
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new;
+
+    $self->{pid}         = undef;
+    $self->{children}    = [];
+    $self->{pidfilename} = 'pvm.pid';
+    $self->{pvmctl}      = '/usr/bin/pvmctl';
+    die "pvmctl not found" unless -x $self->{pvmctl};
+
+    return $self;
+}
+
+sub do_start_vm {
+    my $self = shift;
+    $self->unlink_crash_file;
+    $self->start_lpar();
+    return {};
+}
+
+sub runcmd {
+    diag "running " . join(' ', @_);
+    return system(@_);
+}
+
+sub powercontrol {
+    my $action = $_[0];
+    my $vars   = \%bmwqemu::vars;
+    my $cmd    = "";
+    my $opts   = "";
+    diag "action is $action";
+    if ($action eq 'on') {
+        $cmd = "power-on";
+    }
+    if ($action eq 'off') {
+        $cmd = "power-off";
+    }
+    if ($action eq 'reset') {
+        $cmd = "restart";
+    }
+    runcmd("/usr/bin/pvmctl", "lpar", $cmd, "-i", "name=" . $vars->{LPAR});
+}
+sub attach_console {
+    my ($self, $args) = @_;
+    my $vars    = \%bmwqemu::vars;
+    my $vncport = qx/sudo \/usr\/sbin\/mkvtermutil --id $vars->{LPARID} --vnc --local --log serial0 2>\/dev\/null/;
```

Comment:
this evil quoting looks horrible. You can use qx{} too - using / as seperator with a string with tons of / is kind of crazy